dragonchain/lib/faas.py

```diff
@@ -0,0 +1,45 @@
+import os
+import base64
+import json
+from typing import List, Dict, Optional, cast, Any
+
+import requests
+
+from dragonchain.lib import faas
+from dragonchain import exceptions
+
+FAAS_GATEWAY = os.environ.get("FAAS_LOGS_GATEWAY") or os.environ["FAAS_GATEWAY"]
+
+
+def get_faas_auth() -> str:
+    """Gets authorization to use OpenFaaS
+
+        Returns:
+            A string containing authorization for OpenFaaS.
+    """
+    with open("/etc/openfaas-secret/user", "r") as file:
+        username = file.read()
+    with open("/etc/openfaas-secret/password", "r") as file:
+        password = file.read()
+
+    return f"Basic {base64.b64encode(f'{username}:{password}'.encode('utf-8')).decode('ascii')}"
+
+
+def _get_raw_logs(contract_id: str, since: Optional[str] = None, tail: Optional[int] = 100) -> List[str]:
+    """Calls openfaas /system/logs endpoint with query parameters for a specific contract"""
+    query_params = cast(Dict[str, Any], {"name": f"contract-{contract_id}", "tail": tail, "since": since})
+    headers = {} if os.environ.get("FAAS_LOGS_GATEWAY") else {"Authorization": faas.get_faas_auth()}
```

Comment:
my vote is have `get_faas_auth` simply return an empty string if it errors, rather than throwing an exception, then always include this header. Cleans up this potentially confusing branch which isn't obvious it's for a proxy